web/src/components/routeFiles/FlashcardComponent.tsx

```diff
@@ -42,6 +46,10 @@ const SelectedButton = () => {
             }
             onClick={() => {
                 setSelected(!selected);
+                if (onSelect) {
+                    // remember state doesn't update until next re-render
+                    onSelect(!selected);
```

Comment:
Could be worth doing a `useEffect`? Not sure though

Reply:
Otherwise, might be clearer to do the `onSelect` first, so there is no ambiguity in whether the state has updated